File: MobileTrailer/AboutTrailerViewController.m
#import "AboutTrailerViewController.h"

@implementation AboutTrailerViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	self.versionNumber.text = [@"Version " stringByAppendingString:currentAppVersion];
	self.licenseText.textContainerInset = UIEdgeInsetsMake(0, 10, 10, 10);
}

- (void)viewDidAppear:(BOOL)animated
{
	[super viewDidAppear:animated];
	self.licenseText.contentOffset = CGPointZero;
}

- (IBAction)ipadLink:(UIBarButtonItem *)sender
{
	NSString *url = @"https://github.com/ptsochantaris/trailer";
	[[UIApplication sharedApplication] openURL:[NSURL URLWithString:url]];
}

- (IBAction)done:(UIBarButtonItem *)sender
{
	if(app.preferencesDirty) [app startRefresh];
	[self dismissViewControllerAnimated:YES completion:nil];
}

@end

File: MobileTrailer/AboutTrailerViewController.swift

import UIKit

class AboutTrailerViewController: UIViewController {
	@IBOutlet var versionNumber: UILabel!
	@IBOutlet var licenseText: UITextView!

	override func viewDidLoad() {
		super.viewDidLoad()
        var buildNumber = NSBundle.mainBundle().infoDictionary!["CFBundleVersion"] as String
		versionNumber.text = "\(currentAppVersion) (\(buildNumber))";
		licenseText.textContainerInset = UIEdgeInsetsMake(0, 10, 10, 10)
	}

	override func viewDidAppear(animated: Bool) {
		super.viewDidAppear(animated)
		licenseText.contentOffset = CGPointZero
	}

	@IBAction func linkSelected() {
		UIApplication.sharedApplication().openURL(NSURL(string: "https://github.com/ptsochantaris/trailer")!)
	}

	@IBAction func doneSelected() {
		if app.preferencesDirty { app.startRefresh() }
		dismissViewControllerAnimated(true, completion: nil)
	}
}
